### ISSUE TITLE
~~qTox don't work anymore on OS X~~ Add visible progress to download

### ISSUE THREAD

**Issue by ghost** [qTox/qTox#936]
Hello,

When I installed the latest update, "qtox_sudo" ask for my admin password but after the program exit. I don't know why.

**Comment by zetok** [comment]
I guess #926 broke it @stqism ?

**Comment by stqism** [comment]
@pwnsdx run this for me: `file /Applications/qtox.app/Contents/MacOS/qtox` and then `cat /Applications/qtox.app/Contents/MacOS/updater | shasum`

99% sure the issue is because of github caching something that Jenkins downloads.

**Comment by ghost** [comment]
pwnsdx:~ s$ file /Applications/qtox.app/Contents/MacOS/qtox
/Applications/qtox.app/Contents/MacOS/qtox: Mach-O 64-bit executable x86_64

pwnsdx:~ s$ cat /Applications/qtox.app/Contents/MacOS/updater | shasum
9dd81b5ad9dcc2e4072f86fe9069e8b93ad4fc07  -

pwnsdx:~ s$ /Applications/qtox.app/Contents/MacOS/qtox ; exit;
Settings: Loading from  "/Users/s/Library/Preferences/tox/qtox.ini"
Using embeded bootstrap nodes list
built on:  10:46:45 Dec 13 2014 ( 1418496368 )
commit:  e7cad9b942fd5878138f181bc47f81658452759b 

AutoUpdater::getLocalFlist: Can't open local flist
AutoUpdater::parseflist: Empty data
AutoUpdater: About to start the qTox updater to install a local update
logout

**Comment by ghost** [comment]
`rm -rf /Users/s/Library/Preferences/tox/update/`

Problem fixed, looks like I leave qTox BEFORE the download finished. You should add a percentage and prevent the app from exiting and even if the user exit or the app crash by hazard, check at the boot the shasum of the update and if it is not matching with the update, remove the directory

**Comment by stqism** [comment]
Yeah, it's a qTox bug, not an updater bug.

I do have a way to work around it though.

Tl;Dr: qTox is not downloading the update fully and marking it as so, so it triggers the installer to install a half downloaded update on top of the current install.

**Comment by dubslow** [comment]
@tux3

**Comment by tux3** [comment]
The updater never really marks when a download is complete. It has a list of things to download (the flist) and shouldn't install an update if it isn't complete.

Now qTox tries to resume an update if it isn't completely downloaded, but the code isn't perfect. Is this still reproducible right now ?

**Comment by Tamriel** [comment]
Yes, this is still reproducible.

**Comment by tux3** [comment]
We don't officially support the OS X updater at this point, so this bug doesn't really apply anymore; however we did implement a visible progress bar and I'm happy to say the core updater code works fine on Windows systems, so this should be easily portable to OS X in the future.
